diff --git a/src/server/game/Spells/SpellEffects.cpp b/src/server/game/Spells/SpellEffects.cpp
index a30b6c16a..79f4cf880 100644
--- a/src/server/game/Spells/SpellEffects.cpp
+++ b/src/server/game/Spells/SpellEffects.cpp
@@ -3150,7 +3150,8 @@ void Spell::EffectTameCreature(SpellEffIndex /*effIndex*/)
     if (creatureTarget->IsPet())
         return;
 
-    if (!m_caster->IsClass(CLASS_HUNTER, CLASS_CONTEXT_PET))
+    bool canHybridTame = m_spellInfo->Id == 1515 && m_caster->IsPlayer() && m_caster->ToPlayer()->HasSpell(1515);
+    if (!m_caster->IsClass(CLASS_HUNTER, CLASS_CONTEXT_PET) && !canHybridTame)
         return;
 
     // cast finish successfully
