diff --git a/src/server/game/Entities/Pet/Pet.cpp b/src/server/game/Entities/Pet/Pet.cpp
--- a/src/server/game/Entities/Pet/Pet.cpp
+++ b/src/server/game/Entities/Pet/Pet.cpp
@@ -457,17 +457,28 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
                         break;
                     }
 
+            bool hasPetSpellSlots = false;
+            for (uint8 i = ACTION_BAR_INDEX_PET_SPELL_START; i < ACTION_BAR_INDEX_PET_SPELL_END; ++i)
+                if (UnitActionBarEntry const* ab = m_charmInfo->GetActionBarEntry(i))
+                    if (ab->GetAction())
+                    {
+                        hasPetSpellSlots = true;
+                        break;
+                    }
+
             if (!hasCommandSlots)
+                m_charmInfo->InitPetActionBar();
+
+            if (!hasPetSpellSlots)
             {
-                m_charmInfo->InitPetActionBar();
                 for (PetSpellMap::const_iterator itr = m_spells.begin(); itr != m_spells.end(); ++itr)
                 {
                     if (itr->second.state == PETSPELL_REMOVED)
                         continue;
 
                     SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(itr->first);
                     if (spellInfo && !spellInfo->IsPassive())
                         m_charmInfo->AddSpellToActionBar(spellInfo, itr->second.active);
                 }
             }
