diff --git a/src/server/game/Entities/Pet/Pet.cpp b/src/server/game/Entities/Pet/Pet.cpp
--- a/src/server/game/Entities/Pet/Pet.cpp
+++ b/src/server/game/Entities/Pet/Pet.cpp
@@ -477,7 +477,7 @@ bool Pet::LoadPetFromDB(Player* owner, uint32 petEntry, uint32 petnumber, bool c
                         continue;
 
                     SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(itr->first);
-                    if (spellInfo && !spellInfo->IsPassive())
+                    if (spellInfo && itr->second.active != ACT_PASSIVE)
                         m_charmInfo->SetActionBar(petSpellSlot++, itr->first, itr->second.active);
                 }
             }
